Emby.Server.Implementations/Diagnostics/CommonProcess.cs

```diff
@@ -121,30 +121,29 @@ public bool WaitForExit(int timeMs)
             return _process.WaitForExit(timeMs);
         }
 
-        public Task<bool> WaitForExitAsync(int timeMs)
+        private bool WaitForExitAsyncCompletion(CancellationToken cancellationToken)
         {
-            //if (_process.WaitForExit(100))
-            //{
-            //    return Task.FromResult(true);
-            //}
+            while (!HasExited && !cancellationToken.IsCancellationRequested) { }//wait for exit
```

Comment:
This is just busy waiting. Your CPU is not gonna be happy about it.

Reply:
Yea I see now. I'll guess I'll dive deeper into the code to make a change in HasExited trigger something.